app/main/views/frameworks.py

```diff
@@ -68,21 +68,23 @@ def framework_dashboard(framework_slug):
             )
 
     drafts, complete_drafts = get_drafts(data_api_client, framework_slug)
-
     supplier_framework_info = get_supplier_framework_info(data_api_client, framework_slug)
     declaration_status = get_declaration_status_from_info(supplier_framework_info)
-    supplier_is_on_framework = get_supplier_on_framework_from_info(supplier_framework_info)
-
     # Do not show a framework dashboard for earlier G-Cloud iterations
     if declaration_status == 'unstarted' and framework['status'] == 'live':
         abort(404)
 
-    application_made = supplier_is_on_framework or (len(complete_drafts) > 0 and declaration_status == 'complete')
+    supplier_is_on_framework = get_supplier_on_framework_from_info(supplier_framework_info)
+    application_made = complete_drafts and declaration_status == 'complete'
     lots_with_completed_drafts = [lot for lot in framework['lots'] if count_drafts_by_lot(complete_drafts, lot['slug'])]
 
     framework_dates = content_loader.get_message(framework_slug, 'dates')
     framework_urls = content_loader.get_message(framework_slug, 'urls')
 
+    # Confidence banner.
+    if application_made and framework['status'] == 'open':
+        flash('submission_confidence', 'success')
```

Comment:
So we're flashing a message... to ourselves? Is this a normal pattern in our code? (genuine question)